src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -76,6 +76,10 @@ public WriteAheadJournal(StorageEnvironment env)
                 var currentJournal = journalFilesCount > 0 ? _journalIndex : -1;
                 header->Journal.CurrentJournal = currentJournal;
                 header->Journal.JournalFilesCount = journalFilesCount;
+                if(journalFilesCount == 0)
+                {
+                    header->Journal.LastSyncedJournal = -1;
+                }
```

Comment:
I'm a bit worried about that change. Note we call `_updateLogInfo` action from `UpdateFileHeaderAfterDataFileSync`

https://github.com/ravendb/ravendb/blob/d476f61ccb2ffbc77fe95772c3599183a1204097/src/Voron/Impl/Journal/WriteAheadJournal.cs#L1216-L1226

and this can override potentially `LastSyncedJournal` we set here:

https://github.com/ravendb/ravendb/blob/d476f61ccb2ffbc77fe95772c3599183a1204097/src/Voron/Impl/Journal/WriteAheadJournal.cs#L1221

BTW The case that `JournalFilesCount == 0` might be tricky for us during recovery. See here:
https://github.com/ravendb/ravendb/blob/d476f61ccb2ffbc77fe95772c3599183a1204097/src/Voron/Impl/Journal/WriteAheadJournal.cs#L163-L167
 I'm gonna double check that - I'm worried that we might still have some journals that were not synced and we should to run the recovery